core/transform.h

```diff
@@ -13,6 +13,56 @@ namespace scipp::core {
 
 namespace detail {
 
+template <class T> struct ValueAndVariance {
+  T value;
+  T variance;
+};
+
+template <class T1, class T2>
+constexpr auto operator+(const ValueAndVariance<T1> a,
+                         const ValueAndVariance<T2> b) noexcept {
+  return ValueAndVariance{a.value + b.value, a.variance + b.variance};
+}
+template <class T1, class T2>
+constexpr auto operator-(const ValueAndVariance<T1> a,
+                         const ValueAndVariance<T2> b) noexcept {
+  return ValueAndVariance{a.value - b.value, a.variance - b.variance};
+}
+template <class T1, class T2>
+constexpr auto operator*(const ValueAndVariance<T1> a,
+                         const ValueAndVariance<T2> b) noexcept {
+  return ValueAndVariance{a.value * b.value,
+                          a.variance * b.value * b.value +
+                              b.variance * a.value * a.value};
+}
+template <class T1, class T2>
+constexpr auto operator/(const ValueAndVariance<T1> a,
+                         const ValueAndVariance<T2> b) noexcept {
+  return ValueAndVariance{a.value / b.value,
+                          a.variance / (b.value * b.value) +
+                              b.variance / (a.value * a.value)};
+}
+
+template <class T1, class T2>
+constexpr auto operator+(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  return ValueAndVariance{a.value + b, a.variance};
+}
+template <class T1, class T2>
+constexpr auto operator-(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  return ValueAndVariance{a.value - b, a.variance};
+}
+template <class T1, class T2>
+constexpr auto operator*(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  return ValueAndVariance{a.value * b, a.variance * b * b};
+}
+template <class T1, class T2>
+constexpr auto operator/(const ValueAndVariance<T1> a, const T2 b) noexcept {
+  return ValueAndVariance{a.value / b, a.variance / (b * b)};
```

Comment:
Same comment as above, if `b` is zero

Reply:
Does not apply, since there is now division by `a.value` (since there is no variance on `b`), so the infinity does not show up.